modules/internal/ChapelArray.chpl

```diff
@@ -2395,10 +2465,14 @@ module ChapelArray {
   // integers and ranges; that is, it is a valid argument list for rank
   // change
   proc _validRankChangeArgs(args, type idxType) param {
+    proc _isRange(type idxType, r: range(?)) param return true;
```

Comment:
I'll describe it in the PR message, but it's because this() was ambiguous between element access and rank change, because validRankChangeArgs would return true for arguments of say int type only.